examples/signers/src/example.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * EXAMPLE
+ * Create and use signers with @solana/web3.js.
+ *
+ * Before running any of the examples in this monorepo, make sure to set up a test validator by
+ * running `pnpm test:live-with-test-validator:setup` in the root directory.
+ *
+ * To run this example, execute `pnpm start` in this directory.
+ */
+import { createLogger } from '@solana/example-utils/createLogger.js';
+import {
+    address,
+    appendTransactionMessageInstruction,
+    CompilableTransactionMessage,
+    compileTransaction,
+    createKeyPairSignerFromBytes,
+    createKeyPairSignerFromPrivateKeyBytes,
+    createNoopSigner,
+    createSignableMessage,
+    createSolanaRpc,
+    createTransactionMessage,
+    generateKeyPairSigner,
+    getBase58Decoder,
+    getUtf8Encoder,
+    MessagePartialSigner,
+    partiallySignTransactionMessageWithSigners,
+    pipe,
+    setTransactionMessageFeePayerSigner,
+    setTransactionMessageLifetimeUsingBlockhash,
+    TransactionPartialSigner,
+    TransactionSigner,
+} from '@solana/web3.js';
+import install from '@solana/webcrypto-ed25519-polyfill';
+import { getTransferSolInstruction } from '@solana-program/system';
+import { readFile } from 'fs/promises';
+import path from 'path';
+
+const log = createLogger('Signers');
+
+/**
+ * SETUP: INSTALL WEB CRYPTO POLYFILL
+ * This polyfill may be required to access the Web Crypto API in your environment.
+ */
+// @ts-expect-error FIXME(https://github.com/solana-labs/solana-web3.js/pull/3047)
+install();
+
+/**
+ * SETUP: CREATE A NEW TRANSACTION MESSAGE
+ * This helper function creates a new transaction message for us such that:
+ * - Its lifetime is set to the latest blockhash.
+ * - Its fee payer is set to the given signer.
+ * - It contains a single instruction to transfer 1 SOL to some other address.
+ *
+ * You can read more about this transaction and how to send it
+ * in the `examples/transfer-lamports` example.
+ */
+async function getTransferSolTransactionMessage(signer: TransactionSigner) {
+    // Create the transfer SOL instruction by passing the signer as the source.
+    const instruction = getTransferSolInstruction({
+        amount: 1n,
+        destination: address('ED1WqT2hWJLSZtj4TtTdoovmpMrr7zpkUdbfxmcJR1Fq'),
+        source: signer, // <- We pass the signer here, not just the address.
+    });
+
+    // Get the latest blockhash to use as a transaction lifetime constraint.
+    const rpc = createSolanaRpc('http://127.0.0.1:8899');
+    const { value: latestBlockhash } = await rpc.getLatestBlockhash().send();
```

Comment:
Ooh, is it not preferable to reduce the complexity of this thing by:

1. Removing the comment about setting up the validator from the top of this file
2. Jamming a fake blockhash in here

You'll still be able to sign stuff, but you won't have to set up a validator or import `createSolanaRpc` to this example.

Reply:
Goot shout! I've deleted all requirements for a test validator by using a mock blockhash.